main_aug.py

```diff
@@ -1,38 +1,35 @@
-flags.DEFINE_integer("Img_width", 1024, "The size of the output images to produce. If None, same value as output_height [None]")
-flags.DEFINE_boolean("chromab_flag", True, "flag that specifies whether to perform Chromatic aberration augmentation")
-flags.DEFINE_boolean("blur_flag", True, "flag that specifies whether to perform Blur augmentation")
-flags.DEFINE_boolean("exposure_flag", True, "flag that specifies whether to perform Exposure augmentation")
-flags.DEFINE_boolean("noise_flag", True, "flag that specifies whether to perform noise augmentation")
-flags.DEFINE_boolean("color_flag", True, "flag that specifies whether to perform color shift augmentation")
-flags.DEFINE_boolean("save_aug_params_flag", False, "flag that specifies whether to save aug. parameters for each image")
-#
-flags.DEFINE_string("input_fname_pattern", "*.png", "Glob pattern of filename of input images [*]")
-flags.DEFINE_string("results_dir", "results", "Directory name to save the augmented images [results]")
-FLAGS = flags.FLAGS
+parser = argparse.ArgumentParser(description='Augment a dataset')
+parser.add_argument('-n', type=int, default=1, nargs='?', help='sets the number of augmentations to perform on the dataset i.e., setting n to 2 means the dataset will be augmented twice')
+parser.add_argument('-b', '--batch_size', type=int, default=64, nargs='?', help='size of batches; must be a multiple of n and >1')
+parser.add_argument('-c', '--channels', type=int, default=3, nargs='?', help='dimension of image color channel (note that any channel >3 will be discarded')
+parser.add_argument('-i', '--input', type=str, help='path to the dataset to augment')
+parser.add_argument('-o', '--output', type=str, default='results', nargs='?', help='path where the augmented dataset will be saved')
+parser.add_argument('--pattern', type=str, default="*.png", nargs='?', help='glob pattern of filename of input images')
+parser.add_argument('--image_height', type=int, default=512, nargs='?', help='size of the output images to produce (note that all images will be resized to the specified image_height x image_width)')
+parser.add_argument('--image_width', type=int, default=1024, nargs='?', help='size of the output images to produce. If None, same value as output_height')
+parser.add_argument('--chromatic_aberration', type=bool, default=False, nargs='?', help='perform chromatic aberration augmentation')
+parser.add_argument('--blur', type=bool, default=False, nargs='?', help='perform blur augmentation')
+parser.add_argument('--exposure', type=bool, default=False, nargs='?', help='perform exposure augmentation')
+parser.add_argument('--noise', type=bool, default=False, nargs='?', help='perform noise augmentation')
+parser.add_argument('--colour_shift', type=bool, default=False, nargs='?', help='perform colour shift augmentation')
+parser.add_argument('--save_params', type=bool, default=False, nargs='?', help='save augmentation parameters for each image')
+args = parser.parse_args()
```

Comment:
Replaced tf's deprecated flags with the much more robust argparse. Cleaned up the arguments to hopefully be clearer